### PR TITLE
Request OPTIONS fix to return all allowed methods

### PR DESCRIPTION
**Pull request by michalbundyra**
The integration tests are pulled from zend-expressive-router repository.

See: https://github.com/zendframework/zend-expressive-router/pull/53

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -19,17 +19,23 @@
         "slack": "https://zendframework-slack.herokuapp.com",
         "forum": "https://discourse.zendframework.com/c/questions/expressive"
     },
+    "repositories": [
+        {
+            "type": "git",
+            "url": "https://github.com/webimpress/zend-expressive-router"
```
Comment (geerteltink) [inline composer.json]: I guess this is here for testing only as it should never be merged.

File: composer.json [hunk 2]
```diff
@@ -19,17 +19,23 @@
...
+        }
+    ],
     "require": {
         "php": "^7.1",
         "fig/http-message-util": "^1.1.2",
         "psr/http-message": "^1.0.1",
-        "zendframework/zend-expressive-router": "^3.0.0alpha1 || ^3.0",
+        "zendframework/zend-expressive-router": "dev-hotfix/options-request",
```
Comment (geerteltink) [inline composer.json]: Same, just a reminder to change this before merging.

File: composer.json [hunk 3]
```diff
@@ -19,17 +19,23 @@
...
         "zendframework/zend-psr7bridge": "^0.2.2 || ^1.0.0",
         "zendframework/zend-router": "^3.0.2"
     },
     "require-dev": {
         "malukenho/docheader": "^0.1.6",
-        "phpunit/phpunit": "^6.5.3",
+        "phpunit/phpunit": "^7.0.2",
```
Comment (geerteltink) [inline composer.json:32]: Can we please stop with all these bugfix requirements all the time? Why is 7.0.2 needed? Do the tests fail with ^7.0?

**Review by geerteltink — DISMISSED** [review]

File: composer.json (hunk 1 above)
Comment (michalbundyra) [reply composer.json]: This PR depends on  zendframework/zend-expressive-router#53 and there is no other way (or I don't know it) to test it than including the repository from my branch with changes. Once  zendframework/zend-expressive-router#53 is merged I'll update this (and also other PRs).

File: composer.json (hunk 3 above)
Comment (michalbundyra) [reply composer.json:32]: For all packages we are always using bugfixes version, why we shouldn't do the same for PHPUnit? It's the latest version available, and it doesn't matter too much as it is just testing framework. I wouldn't bother about it at all.

Comment (weierophinney) [reply composer.json:32]: I'm with @xtreamwayz here; if you're not updating the requirements for a specific reason, there's no need to bump them typically.

(That said, I won't block this PR based on that; just noting that we shouldn't be bumping requirements on every patch, but only when we really need to.)

File: test/IntegrationTest.php [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+ * @copyright Copyright (c) 2018 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive-zendrouter/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace ZendTest\Expressive\Router;
+
+use Zend\Expressive\Router\RouterInterface;
+use Zend\Expressive\Router\Test\IntegrationTest as RouterIntegrationTest;
```
Comment (weierophinney) [inline test/IntegrationTest.php]: I renamed the test case when I merged the patch on zend-expressive-router; it's now `ImplicitMethodsIntegrationTest`.

**Pull request merged**